x-pack/test/token_api_integration/auth/index.js

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export default function ({ loadTestFile }) {
+  describe('token-based auth', function () {
+    this.tags('ciGroup6');
+    loadTestFile(require.resolve('./login'));
```

Comment:
We're only testing the login, and logout functionality at the moment. We're missing coverage for when the `Authorization: Bearer` token is passed via a header, in addition to ensuring subsequent requests using the session cookie work, and refresh token tests.

Reply:
To what extent are you looking to integration test the token provider specifically?

The unit tests verify that header auth results in headers getting wired up to send off to Elasticsearch. So we'll want an integration test that verifies the assumptions we make in that test around shape and contents of the header.

The unit tests verify that an expired token request results in auth via a refresh token process. So we'll want an integration test that verifies the assumptions we make about the shape, contents, and error responses of that refresh process.

Regarding subsequent requests, is there not already a test for the authentication system in general that verifies that state is serialized and returned in a set-cookie header? The token provider unit tests verify that we're returning the appropriate state, which is functionality I assumed we already had tests for. I don't think this benefits from an integration test because we're not really integrating with anything.

I can implement whatever we need. Can you think of anything else?

Reply:
We're in an interesting position with the auth providers, except for the basic auth provider, where the only automated coverage that we have using them, outside of the unit tests, is with these api integration tests. Additionally, since they're highly dependent on the assumptions that we make about how ES behaves in a number of regards, the only assurance that we have that they actually work is using these API integration tests, or manual testing. 

I'd ideally like to see api integration tests for the following scenarios:
- presenting a Bearer token in the Authorization header authenticates the user
- using the session cookie when we get from Kibana to authenticate a subsequent request (this one I'm less concerned about)
- using the session cookie which we get from Kibana to authenticate after the token expiration period passes